packages/orchestrator/internal/server/mem_sandbox_store.go

```diff
@@ -0,0 +1,48 @@
+package server
+
+import (
+	"github.com/e2b-dev/infra/packages/orchestrator/internal/sandbox"
+	"github.com/e2b-dev/infra/packages/shared/pkg/smap"
+)
+
+type MemSandboxStore struct {
+	server    *server
+	sandboxes *smap.Map[*sandbox.Sandbox]
+}
+
+func NewMemSandboxStore(server *server) *MemSandboxStore {
+	return &MemSandboxStore{
+		server:    server,
+		sandboxes: smap.New[*sandbox.Sandbox](),
+	}
+}
+
+func (s *MemSandboxStore) Get(id string) (*sandbox.Sandbox, bool) {
+	sbx, ok := s.sandboxes.Get(id)
+	if !ok {
+		return nil, false
+	}
+	sbx = sbx.WithExternalLogger(s.server.externalLogger)
```

Comment:
setting the logger on sandbox when we creates it makes it a O(1) work for the programmer instead of O(n) when you have to set this in every server handler